"[WinError 5] Access is denied" on Windows when deleting .git directory

I'm unable to use the "github:" fragment on Windows 11, due to a permissions issue that arises consistently when deleting .git. I don't have any unusual setup in my Temp directory, so I would assume that this impacts all Windows users. Despite this error when attempting to delete .git, the entire temporary directory is always successfully cleaned up by Python. 

I can send a PR to fix.

```
> llm -f github:simonw/llm-fragments-github "what does this do and how does it work"
Error: Could not load fragment github:simonw/llm-fragments-github: Error processing repository https://github.com/simonw/llm-fragments-github.git: [WinError 5] Access is denied: 'C:\\Users\\james\\AppData\\Local\\Temp\\tmpl896i092\\.git\\objects\\pack\\pack-5b4427768abe01ac9e14b7188b8cf801244aa651.idx'
```

I got a failure in CI:
```
            except Exception as e:
                # Handle other errors
>               raise ValueError(f"Error processing repository {repo_url}: {str(e)}")
E               ValueError: Error processing repository https://github.com/simonw/test-repo-for-llm-fragments-github.git: [WinError 5] Access is denied: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpb3t706s7\\.git\\objects\\pack\\pack-0697ed0500d3043dc37dda88343bb9e5943db146.idx'
llm_fragments_github.py:87: ValueError
=========================== short test summary info ===========================
FAILED tests/test_fragments_github.py::test_github_loader - ValueError: Error processing repository https://github.com/simonw/test-repo-for-llm-fragments-github.git: [WinError 5] Access is denied: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpb3t706s7\\.git\\objects\\pack\\pack-0697ed0500d3043dc37dda88343bb9e5943db146.idx'
========================= 1 failed, 2 passed in 2.55s =========================
```
Let's see if the PR fixes it.

--- a/llm_fragments_github.py
+++ b/llm_fragments_github.py
@@ -4,7 +4,6 @@ import llm
 import os
 import pathlib
 import re
-import shutil
 import subprocess
 import tempfile
 from urllib.parse import urlparse
@@ -51,32 +50,34 @@ def github_loader(argument: str) -> List[llm.Fragment]:
                 cwd=temp_dir,
             )
 
-            # Remove the .git directory if it still exists
-            git_dir = pathlib.Path(temp_dir) / ".git"
-            if git_dir.exists():
-                shutil.rmtree(git_dir)
-
             # Process the cloned repository
             repo_path = pathlib.Path(temp_dir)
             fragments = []
 
-            # Walk through all files in the repository
-            for file_path in repo_path.glob("**/*"):
-                if file_path.is_file():
-                    try:
-                        # Try to read the file as UTF-8
-                        content = file_path.read_text(encoding="utf-8")
+            # Walk through all files in the repository, excluding .git directory
+            for root, dirs, files in os.walk(repo_path):
+                # Remove .git from dirs to prevent descending into it
+                if ".git" in dirs:
+                    dirs.remove(".git")
 
-                        # Create a relative path for the fragment identifier
-                        relative_path = file_path.relative_to(repo_path)
+                # Process files
+                for file in files:
+                    file_path = pathlib.Path(root) / file
+                    if file_path.is_file():
+                        try:
+                            # Try to read the file as UTF-8
+                            content = file_path.read_text(encoding="utf-8")
 
-                        # Add the file as a fragment
-                        fragments.append(
-                            llm.Fragment(content, f"{argument}/{relative_path}")
-                        )
-                    except UnicodeDecodeError:
-                        # Skip files that can't be decoded as UTF-8
-                        continue
+                            # Create a relative path for the fragment identifier
+                            relative_path = file_path.relative_to(repo_path)
+
+                            # Add the file as a fragment
+                            fragments.append(
+                                llm.Fragment(content, f"{argument}/{relative_path}")
+                            )
+                        except UnicodeDecodeError:
+                            # Skip files that can't be decoded as UTF-8
+                            continue
 
             return fragments
         except subprocess.CalledProcessError as e:

--- a/tests/test_fragments_github.py
+++ b/tests/test_fragments_github.py
@@ -4,7 +4,10 @@ import pytest
 
 def test_github_loader():
     fragments = github_loader("simonw/test-repo-for-llm-fragments-github")
-    assert [(fragment.source, str(fragment)) for fragment in fragments] == [
+    normalized = [
+        (fragment.source.replace("\\", "/"), str(fragment)) for fragment in fragments
+    ]
+    assert normalized == [
         (
             "simonw/test-repo-for-llm-fragments-github/README.md",
             "# test-repo-for-llm-fragments-github\nUsed by tests for https://github.com/simonw/llm-fragments-github\n",

